packages/lib/src/vis/line/DataCurve.tsx

```diff
@@ -1,14 +1,21 @@
 import type { NumArray } from '@h5web/shared';
-import { extend, useThree } from '@react-three/fiber';
 import type { Object3DNode, ThreeEvent } from '@react-three/fiber';
-import { useCallback, useLayoutEffect, useState } from 'react';
+import { extend, useThree } from '@react-three/fiber';
+import { useCallback, useLayoutEffect, useMemo } from 'react';
 import { BufferGeometry, Line } from 'three';
 
+import { useVisCanvasContext } from '../shared/VisCanvasProvider';
+import { createBufferAttr } from '../utils';
 import ErrorBars from './ErrorBars';
 import GlyphMaterial from './GlyphMaterial';
-import { useCanvasPoints } from './hooks';
+import { useValueToErrorPositions, useValueToPosition } from './hooks';
 import { CurveType, GlyphType } from './models';
 
+/* Render points with NaN/Infinity coordinates (i.e. values <= 0 in log)
+ * at origin to avoid Three warning, and outside of camera's field of view
+ * to hide them and any segments connecting them. */
+const CAMERA_FAR = 1000; // R3F's default
```

Comment:
This constant and comment will eventually move into a dedicated subclass that extends `BufferGeometry`. This class will provide an abstraction for initialising geometries and updating buffer attributes, so it should hugely clean up the `useMemo` and `useLayoutEffect` code below.